pkgs/build-support/portable-service/default.nix

```diff
@@ -0,0 +1,75 @@
+{ pkgs, lib, stdenv }:
+/*
+  Create a systemd portable service image
+  https://systemd.io/PORTABLE_SERVICES/
+
+  Example:
+  …
+*/
+{ name
+, version ? "dev"
+, description ? null
+, homepage ? null
+, contents ? [ ]
+, symlinks ? [ ]
+, units ? [ ]
+, squashfsTools ? pkgs.squashfsTools
+, squash-compression ? "xz -Xdict-size 100%"
+, squash-block-size ? "1M"
+}:
+
+let
+  image-name = "${name}_${version}";
+
+  os-release-params = lib.filterAttrs (n: v: v != null) {
+    PORTABLE_ID = name;
+    PORTABLE_PRETTY_NAME = description;
+    HOME_URL = homepage;
+    ID = "nixos";
+    PRETTY_NAME = "NixOS";
+    BUILD_ID = "rolling";
+  };
+
+  os-release = pkgs.writeText "os-release" (lib.generators.toKeyValue {} os-release-params);
+
+  rootfs = stdenv.mkDerivation {
+    name = "rootfs";
+    buildCommand = ''
+      # file system layout
+      mkdir -p $out/etc/systemd/system $out/proc $out/sys $out/dev $out/run $out/tmp $out/var/tmp
+
+      # empty files to mount over with host's version
+      touch $out/etc/resolv.conf $out/etc/machine-id
+
+      # required for portable services
+      cp ${os-release} $out/etc/os-release
+
+      # units **must** be copied to /etc/systemd/system/
+      ${lib.forEach units (u: "cp ${u} $out/etc/systemd/system/${u.name};")}
```

Comment:
I guess?

```suggestion
      ${lib.forEach units (u: "cp ${u.name} $out/etc/systemd/system/${u.name};")}
```

Reply:
so (I wish there were type annotations in Nix 🙂) `units` is expected to be a list of derivations. The derivation has a name attribute, but when used as ${u} in a shell/string context, it will evaluate to the actual path in the nix store.

```
nix-repl> wallabag-service = pkgs.substituteAll { name = "wallabag.service"; src = ./fajl; world = "test"; }

nix-repl> wallabag-service
«derivation /nix/store/4wrl4jsl5li29sk9bry96f7s14imcg25-wallabag.service.drv»

nix-repl> wallabag-service.name
"wallabag.service"

nix-repl> "${wallabag-service}" 
"/nix/store/0674kr7idpjqb89maja8f6vlgzfxcjpv-wallabag.service"

nix-repl> "${wallabag-service.name}"
"wallabag.service"
```